packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeActions.ts

```diff
@@ -78,14 +79,14 @@ export class TreeActions {
             }),
             switchMap((props) =>
               this._loader.loadNodes(props.loadOptions).pipe(
-                collectTreePartsUntil(this._reset, props.initialRootNode),
+                collectTreePartsUntil(this._reset, props.loadOptions.parent, props.initialRootNode),
                 tap({
                   next: (newModel: TreeModel) => {
                     const childNodes = newModel.parentChildMap.get(props.parentId);
                     const firstChildNode = childNodes?.length ? newModel.idToNode.get(childNodes[0]) : undefined;
                     this.handleLoadedHierarchy(props.parentId, newModel);
                     // only report load duration if no error occurs
-                    if (!(firstChildNode && isTreeModelInfoNode(firstChildNode))) {
+                    if (!(firstChildNode && firstChildNode.error)) {
```

Comment:
the error's are now on the parent rather than the first child, right?

Reply:
but now we're going to report load durations even on errors?

Reply:
In this case there would not be any children as the parent was changed, so the if inside would not be hit.